plugins/logs/plugin.go

```diff
@@ -179,16 +178,20 @@ func (p *Plugin) Log(ctx context.Context, decision *server.Info) {
 		Timestamp:   decision.Timestamp,
 	}
 
-	if err := json.NewEncoder(&buf).Encode(event); err != nil {
-		p.logError("Log serialization failed: %v.", err)
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	result, err := p.enc.Write(event, p.buffer)
+	if err != nil {
+		p.logError("Log encoding failed: %v.", err)
 		return
 	}
 
-	p.mtx.Lock()
-	defer p.mtx.Unlock()
-	dropped := p.buffer.Push(buf.Bytes(), false)
-	if dropped > 0 {
-		p.logInfo("Dropped %v events from buffer. Reduce reporting interval or increase buffer size.")
+	if result != nil {
+		dropped := p.buffer.Push(result)
```

Comment:
I'd refactor these lines into a separate function as they're duplicated in oneShot below. For example:

```golang
if result != nil {
  p.bufferChunk(result)
}
```

And then...


```golang
func (p *Plugin) bufferChunk(bs []byte) {
  dropped := p.buffer.Push(bs)
  if dropped > 0 {
     p.logInfo("Dropped %v events...", ...)
  }
}
```